core/commonMain/src/kotlinx/serialization/internal/Tagged.kt

```diff
@@ -159,7 +158,7 @@ public abstract class TaggedEncoder<Tag : Any?> : Encoder, CompositeEncoder {
         tagStack.add(name)
     }
 
-    protected fun popTag(): Tag =
+    private fun popTag(): Tag =
```

Comment:
AFAIR, this method is `protected` because one of the formats authors asked us about it (XML or YAML, I don't remember actually). I know that we do not intend to support these interfaces for such a usage, however, we do not have viable alternative for now besides 'copy-pasting all TaggedEncoder class to your format'. So let's check if this is really true

Reply:
Seems there's BSON for now: https://github.com/agcom/bson/blob/bb58e4259518257de212388d83f07935417f67ca/serialization/src/main/kotlin/com/github/agcom/bson/serialization/decoders/BsonTreeInput.kt#L66

Reply:
Sure, let's return that